src/libbson/src/bson/bson.c

```diff
@@ -1503,14 +1504,26 @@ bson_append_utf8 (bson_t *bson, const char *key, int key_length, const char *val
    HANDLE_KEY_LENGTH (key, key_length);
 
    if (BSON_UNLIKELY (length < 0)) {
-      length = (int) strlen (value);
+      len_sz = strlen (value);
+      if (len_sz > (size_t) INT_MAX) {
+         /* The return of 'false' would allow for possible overflow later on */
+         return bson_append_null (bson, key, key_length);
+      }
+      length = (int) len_sz;
    }
 
-   length_le = BSON_UINT32_TO_LE (length + 1);
+   if (!bson_cmp_less_su (length, UINT32_MAX)) {
+      return false;
+   }
+   length_le = BSON_UINT32_TO_LE ((uint32_t) length + 1u);
+
+   if ((uint32_t) key_length > UINT32_MAX - 1u - 1u - 4u - (uint32_t) length - 1u) {
```

Comment:
```suggestion
   const int64_t num_bytes = INT64_C (1) + key_length + UINT64_C (1) + INT64_C (4) + length + INT64_C (1);

   if (bson_cmp_less_us (UINT32_MAX, num_bytes)) {
```
Reduce redundancy of arithmetic expressions. Note `INT64_C` is required for correctness.

Reuse this variable in `_bson_append` call below:
```c
return _bson_append (bson, 6u, (uint32_t)num_bytes, 1u, ...)
```

Reply:
I was not aware of the `UINT#_C` and `INT#_C` macros. Change incoming.